### ISSUE TITLE
Right-to-left numbering of existential variables in `instantiate` is strange and inconsistent with other features

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#13689]
Re: instantiate (num := term)

I believe we generally count items from left to right, for example in `occurrences` clauses.  It is unexpected that instantiate counts them the opposite way, from right to left.  Is it worth the disruption to users to changes this?  Perhaps we could make positive numbers to count left to right (not backward compatible but consistent with `occurrences`) but also allow negative numbers to count from right to left to make the user updates easy (just add a minus sign).

**Comment by Zimmi48** [comment]
On `instantiate`: I don't think it's worth changing because this variant is *strongly* discouraged anyway.

On negative occurrence counting, if I'm not mistaken it's already supported in standard occurrence clauses.

**Comment by jfehrle** [comment]
Yes, occurrences allows negative numbers.

If the variant is _strongly_ discouraged then the tactic really worth keeping?  (For occurrences there are no names to refer to, so we live with it)

**Comment by Zimmi48** [comment]
Yes, because when you are in the middle of a proof, it may be costly to modify it in a way that allows you to refer to an existential variable by a name.

**Comment by RalfJung** [comment]
> Yes, because when you are in the middle of a proof, it may be costly to modify it in a way that allows you to refer to an existential variable by a name.

How do you do that anyway? Evars do have names when rendered in the goal, but is there any way to use that name for `instantiate`?

**Comment by Zimmi48** [comment]
Not if you didn't set that name yourself. See the doc:
- https://coq.inria.fr/refman/proof-engine/tactics.html#coq:tacn.instantiate
- https://coq.inria.fr/refman/language/extensions/evars.html#existential-variables

See also: https://coq.inria.fr/refman/proofs/writing-proofs/proof-mode.html#coq:tacn.{
